test/test_losses.py

```diff
@@ -9,6 +9,71 @@
 from torch.testing import assert_allclose
 
 
+class TestFocalLossWithLogits:
+    def test_smoke_none(self, device):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2).to('cpu')
+        labels = torch.rand(2, 3, 2) * num_classes
+        labels = labels.to(device).long()
+
+        assert kornia.losses.binary_focal_loss_with_logits(
+            logits, labels, alpha=0.5, gamma=2.0, reduction="none"
+        ).shape == (2, 3, 2)
+
+    def test_smoke_sum(self, device):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2).to('cpu')
+        labels = torch.rand(2, 3, 2) * num_classes
+        labels = labels.to(device).long()
+
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="sum"
+            ).shape == ()
+        )
+
+    def test_smoke_mean(self, device):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2).to('cpu')
+        labels = torch.rand(2, 3, 2) * num_classes
+        labels = labels.to(device).long()
+
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="mean"
+            ).shape == ()
+        )
+
+    def test_smoke_mean_flat(self, device):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2).to('cpu')
+        labels = torch.rand(2, 3, 2) * num_classes
+        labels = labels.to(device).long()
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="mean"
+            ).shape == ()
+        )
+
+    # TODO: implement me
+    def test_jit(self, device):
+        pass
```

Comment:
could you try to implement this one too ? see an example here: https://github.com/kornia/kornia/blob/master/test/geometry/test_linalg.py#L82 (omit lines 83-85)